deploy/cdk/lib/user-content/index.ts

```diff
@@ -0,0 +1,119 @@
+import apigateway = require('@aws-cdk/aws-apigateway');
+import dynamodb = require('@aws-cdk/aws-dynamodb');
+import lambda = require('@aws-cdk/aws-lambda');
+import ssm = require('@aws-cdk/aws-ssm');
+import cdk = require('@aws-cdk/core');
+
+export interface UserContentStackProps extends cdk.StackProps {
+  readonly lambdaCodePath: string
+  readonly tokenAudiencePath: string
+  readonly tokenIssuerPath: string
+};
+
+export class UserContentStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props: UserContentStackProps) {
+    super(scope, id, props);
+
+    // Dynamo Tables
+    const userDynamoTable = new dynamodb.Table(this, 'UsersTable', {
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      partitionKey: {
+        name: 'userName',
+        type: dynamodb.AttributeType.STRING
+      },
+      tableName: `${this.stackName}-users`,
+      removalPolicy: cdk.RemovalPolicy.DESTROY, // NOT recommended for production code
+    });
+
+    const collectionDynamoTable = new dynamodb.Table(this, 'CollectionsTable', {
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      partitionKey: {
+        name: 'uuid',
+        type: dynamodb.AttributeType.STRING
+      },
+      tableName: `${this.stackName}-collections`,
+      removalPolicy: cdk.RemovalPolicy.DESTROY, // NOT recommended for production code
+    });
+    collectionDynamoTable.addGlobalSecondaryIndex({
+      indexName: 'userName',
+      partitionKey: {
+        name: 'userName',
+        type: dynamodb.AttributeType.STRING
+      }
+    });
```

Comment:
This is probably fine, but I wondered if you've looked at using the userName as a sortKey instead of a secondary index? You can query a sort key as well.

Reply:
I did look into using a sort key, but the recommendation seemed to be using a secondary index over a sort key when you needed to frequently query the data without knowing the primary key. I don't know if that's completely accurate, but it seems to work 🤷‍♂